### PR TITLE
fix styles not applied on ssr a/b testing

### PR DESCRIPTION
**Pull request by teleaziz**

### CODE REVIEW

File: src/html.tsx [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+const postProcess = (body: string) => {
+  let globalStyles = '';
+
+  if (body.includes('<template')) {
+    const $ = cheerio.load(body);
+    const templates = $('template');
+
+    templates.toArray().forEach(element => {
+      const str = $(element).html();
+      const styles = cheerio.load(String(str))('style');
```
Comment (mandx) [inline src/html.tsx:20]: Why is `String` needed?
```suggestion
      const styles = cheerio.load(str)('style');
```

File: src/html.tsx [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+        {props.preBodyComponents}
+        <style dangerouslySetInnerHTML={{ __html: globalStyles }}></style>
+        <div key="body" id="___gatsby" dangerouslySetInnerHTML={{ __html: body }} />
+        {props.postBodyComponents}
+      </body>
+    </html>
+  );
+};
+
+export default HTML;
```
Comment (mandx) [inline src/html.tsx:58]: Is this component automatically picked up and used by Gatsby? We don't have to "register"/"import" it?

**Review by mandx — APPROVED** [review]

File: src/html.tsx [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+  return { body, globalStyles };
+};
+
+interface HTMLProps {
+  htmlAttributes: {};
+  headComponents: ReactNode[];
+  bodyAttributes: {};
+  preBodyComponents: ReactNode[];
+  body: string;
+  postBodyComponents: ReactNode[];
```
Comment (mandx) [inline src/html.tsx:36]: `ReactNode` can also be arrays; isn't enough just use `ReactNode` directly?
```suggestion
  headComponents: ReactNode;
  bodyAttributes: {};
  preBodyComponents: ReactNode;
  body: string;
  postBodyComponents: ReactNode;
```

File: src/html.tsx (hunk 1 above)
Comment (teleaziz) [reply src/html.tsx:20]: Casting to make ts happy, str can be null.

File: src/html.tsx (hunk 2 above)
Comment (teleaziz) [reply src/html.tsx:58]: yes, https://www.gatsbyjs.com/docs/custom-html/

File: src/html.tsx (hunk 3 above)
Comment (teleaziz) [reply src/html.tsx:36]: following the ejected types from gatsby https://github.com/gatsbyjs/gatsby/blob/master/packages/gatsby/cache-dir/default-html.js#L31

**Pull request merged**